sdk/go/common/workspace/paths.go

```diff
@@ -93,34 +93,19 @@ func DetectProjectPath() (string, error) {
 
 // DetectProjectStackPath returns the name of the file to store stack specific project settings in. We place stack
 // specific settings next to the Pulumi.yaml file, named like: Pulumi.<stack-name>.yaml
-func DetectProjectStackPath(stackName tokens.QName) (string, error) {
+func DetectProjectStackPath(stackName tokens.QName) (*Project, string, error) {
 	proj, projPath, err := DetectProjectAndPath()
 	if err != nil {
-		return "", err
+		return nil, "", err
 	}
 
 	fileName := fmt.Sprintf("%s.%s%s", ProjectFile, qnameFileName(stackName), filepath.Ext(projPath))
 
-	// Back compat: StackConfigDir used to be called Config.
```

Comment:
We still need to support this. People could still have projects using "config: some_path". Ripping this out is another 4.0 change (maybe we should start leaving "TODO(4.0)" comments around for stuff like this)

Reply:
Brought back support for defining `config: some_path` by rewriting it to `stackConfigDir` when it is a string. Erroring out when both are defined ✅